pkg/server/getRecommendations.go

```diff
@@ -0,0 +1,144 @@
+package server
+
+import (
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"strings"
+
+	"github.com/golang/gddo/httputil/header"
+	"github.com/stolostron/recommends/pkg/config"
+	"github.com/stolostron/recommends/pkg/helpers"
+	"github.com/stolostron/recommends/pkg/utils"
+
+	klog "k8s.io/klog/v2"
+)
+
+var list_recommendations_url = config.Cfg.KruizeURL + "/listRecommendations"
+
+//struct to capture input for getRecommendations request
+type RecommendationInput []struct {
+	RecommendationId string `json:"recommendation_id"`
+	Namespace        string `json:"namespace"`
+	ClusterName      string `json:"cluster_name"`
+}
+
+var requestUrlList []string
+
+func getRecommendations(w http.ResponseWriter, r *http.Request) {
+	klog.Infof("Received Request for list Recommendations")
+
+	var getRecommendations RecommendationInput
+	var recommendations []ListRecommendations
+
+	//check content type is json
+	if r.Header.Get("Content-Type") != "" {
+		value, _ := header.ParseValueAndParams(r.Header, "Content-Type")
+		if value != "application/json" {
+			msg := "Content-Type header is not application/json"
+			http.Error(w, msg, http.StatusUnsupportedMediaType)
+			return
+		}
+	}
+
+	//decode user's input:
+	dec := json.NewDecoder(r.Body)
+	err := dec.Decode(&getRecommendations)
+
+	if ok := helpers.ErrorHandlingRequests(w, err); !ok {
+		return
+	}
+
+	recommendationId := getRecommendations[0].RecommendationId
+	namespace := getRecommendations[0].Namespace
+	clusterName := getRecommendations[0].ClusterName
+
+	//if recommendationid provided default to use recommendation id
+	if recommendationId != "" {
+		reqParts := strings.Split(recommendationId, "_")
+		id := reqParts[3]
+
+		buildKruizeListRequest(id)
+
+	}
+	//if recommendationid is missing but clustername and namespace provided:
+	if namespace != "" && clusterName != "" && recommendationId == "" {
+		clusterNamespace := clusterName + "_" + namespace
+		id := RecommendationMap.ClusterNamespace[clusterNamespace]
+
+		buildKruizeListRequest(id)
+
+		// if missing both namespace and recommendation or if missing both namespace and clustername error
+	} else if namespace == "" && recommendationId == "" || namespace == "" && clusterName == "" {
+		klog.V(5).Infof("Request missing both RecommendationId and Namespace. Need at least one to fulfill request.")
+		http.Error(w, "{\"message\":\"Both  RecommendationId and Namespace cannot be empty.\"}", http.StatusBadRequest)
+		return
+	}
+
+	klog.Info(requestUrlList)
```

Comment:
Please add a message to provide context for this log and consider a higher verbosity.